core/object/object.cpp

```diff
@@ -165,6 +165,37 @@ MethodInfo MethodInfo::from_dict(const Dictionary &p_dict) {
 	return mi;
 }
 
+uint32_t MethodInfo::get_compatibility_hash() const {
+	bool has_return = (return_val.type != Variant::NIL) || (return_val.usage & PROPERTY_USAGE_NIL_IS_VARIANT);
+
+	uint32_t hash = hash_murmur3_one_32(has_return);
+	hash = hash_murmur3_one_32(arguments.size(), hash);
+
+	if (has_return) {
+		hash = hash_murmur3_one_32(return_val.type, hash);
+		if (return_val.class_name != StringName()) {
+			hash = hash_murmur3_one_32(return_val.class_name.operator String().hash(), hash);
+		}
+	}
+
+	for (const PropertyInfo &arg : arguments) {
+		hash = hash_murmur3_one_32(arg.type, hash);
+		if (arg.class_name != StringName()) {
+			hash = hash_murmur3_one_32(arg.class_name.operator String().hash(), hash);
```

Comment:
`StringName` has a `hash()` function that I think is guaranteed to be the same as `.operator String().hash()`.

Reply:
Hm, looking through the code, this _seems_ to be true.

Depending on how a `StringName` is created, it looks like it'll either be hashed via `string.hash()` (which would be exactly the same) or `String::hash(const_char_ptr)` which has code that looks to be the same, but it makes me a little nervous given that the former is using `uint32_t` for the characters and the latter is using `uint8_t`.

In any case, it does seem like they should be the same!

Reply:
I _think_ we can be sure of this fact since `StringName` uses the hash to identify itself. But yeah, since this is not at all performance critical, I'll leave it to you if you want to keep it as-is.

Reply:
I've swapped this for `StringName`'s `hash()` in my latest push. Thanks!